Foundation/NSNotification.swift

```diff
@@ -57,32 +58,43 @@ private let _defaultCenter: NSNotificationCenter = NSNotificationCenter()
 
 public class NSNotificationCenter : NSObject {
     
-    private var observers: [NSNotificationReceiver]
-    
-    public required override init() {
-        observers = [NSNotificationReceiver]()
-    }
+    private var _observers = [NSNotificationReceiver]()
+    private let _observersLock = NSLock()
     
     public class func defaultCenter() -> NSNotificationCenter {
         return _defaultCenter
     }
     
     public func postNotification(notification: NSNotification) {
+
         let name = notification.name
         let sender = notification.object
-        
-        let sendTo = observers.filter { observer in
+
+        _observersLock.lock()
+        let sendTo = _observers.filter { observer in
             let sameName = (observer.name == nil || observer.name == name)
             let sameSender = (observer.sender == nil || observer.sender === sender)
             
             return sameSender && sameName
         }
```

Comment:
I would propose to use `defer` statement for releasing the lock. Just to be sure that the lock is released even if the exception exits prematurely.